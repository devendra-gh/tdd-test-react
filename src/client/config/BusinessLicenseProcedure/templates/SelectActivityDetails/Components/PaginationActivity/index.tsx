import React from 'react';
import { IVariables } from '@tamm/app-composer';
import Pagination from '@tamm/ui-lib-v2-pagination';
import { noRefCheck } from 'client/config/BusinessLicenseProcedure/utils';

function PaginationActivity(props: IVariables) {
  const { formSelectActivity } = props;
  const {
    activitiesCurrentPage,
    activitiesRecInPage,
    activitiesTotalCount,
    showTable,
  } = formSelectActivity;
  return (
    <>
      {activitiesTotalCount / activitiesRecInPage > 1 && showTable && (
        <>
          <Pagination
            current={activitiesCurrentPage}
            disabled={false}
            i18n={noRefCheck}
            pageSize={activitiesRecInPage}
            total={activitiesTotalCount}
            onChange={(currentPage: number) => {
              props.onChange('page', currentPage, props);
            }}
          />
        </>
      )}
    </>
  );
}

export default PaginationActivity;
