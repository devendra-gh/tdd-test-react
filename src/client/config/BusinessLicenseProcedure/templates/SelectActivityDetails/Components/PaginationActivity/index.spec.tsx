import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router';
import { IVariables } from '@tamm/app-composer';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import PaginationActivity from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('PaginationActivity', () => {
  let props: IVariables;

  beforeEach(() => {
    props = {
      onChange: jest.fn(),
      history: {
        push: jest.fn(),
      },
      actions: {
        stepsStatus: {
          update: jest.fn(),
        },
        formSelectActivity: {
          update: jest.fn(),
        },
      },
      getInitialState: jest.fn(),
      formSelectActivity: {
        activitiesCurrentPage: 1,
        activitiesRecInPage: 1,
        activitiesTotalCount: 2,
        showTable: true,
      },
      onSubmit: jest.fn(),
      i18n: (key: string) => key,
      title: 'FormTemplate',
      subTitle: 'subTitle',
      description: 'desc',
      locale: 'en',
      currentStep: 'economic_licence',
      currentSubStep: 'payment',
      buttons: [
        {
          label: 'btn',
          onClick: jest.fn(),
        },
      ],
      steps: [],
      stepStatus: {
        economic_name: 'economic_name',
        'economic_name.payment': 'economic_name.payment',
      },
      list: {
        labelHeading: 'label',
        valueHeading: 'value',
        listDetails: {
          label: 'label',
          value: 'value',
        },
      },
      totalSection: 1000,
    };
  });

  it('should show pagination', async () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <PaginationActivity {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(getByText('1'));
  });
});
