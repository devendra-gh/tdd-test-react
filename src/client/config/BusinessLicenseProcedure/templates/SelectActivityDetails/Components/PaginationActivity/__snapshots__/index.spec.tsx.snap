// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationActivity should show pagination 1`] = `
<div>
  <div
    class="ui-lib-viewport ui-lib-viewport_lg"
  >
    <div
      class="ui-lib-pagination"
    >
      <div
        aria-label=""
        class="ui-lib-pagination__item ui-lib-pagination__item_page ui-lib-pagination__item_current"
        role="button"
        tabindex="0"
      >
        1
      </div>
      <div
        aria-label=""
        class="ui-lib-pagination__item ui-lib-pagination__item_page"
        role="button"
        tabindex="0"
      >
        2
      </div>
    </div>
  </div>
</div>
`;
